test/lua_test.exs

```diff
@@ -395,6 +395,38 @@ defmodule LuaTest do
                return global, success, message
                """)
     end
+
+    test "it can be given a max_reductions option" do
+      assert {:done, ["done"], _} =
+               Lua.eval!("return \"done\" ", max_reductions: :infinity)
+
+      assert {:max_reductions_hit, _} =
+               Lua.eval!(
+                 """
+                 local a = 1
+                 b = a + 1
+                 c = b + 1
+                 return "done"
+                 """,
+                 max_reductions: 1
+               )
+    end
+
+    test "it can be given a max_instructions option" do
+      assert {:done, ["done"], _} =
+               Lua.eval!("return \"done\" ", max_instructions: :infinity)
+
+      assert {:max_instructions_hit, _} =
+               Lua.eval!(
+                 """
+                 local a = 1
+                 b = a + 1
+                 c = b + 1
+                 return "done"
+                 """,
+                 max_instructions: 1
+               )
+    end
```

Comment:
These currently break as we don't have the correct return value yet--but, stubbing that in would add too many hacks for the moment so I just left these ready.